src/west/app/project.py

```diff
@@ -155,19 +156,61 @@ def __init__(self):
             'init',
             'create a west workspace',
             f'''\
-Creates a west workspace.
-
-With -l, creates a workspace around an existing local repository;
-without -l, creates a workspace by cloning a manifest repository
-by URL.
+Initialize a west workspace, either from a remote manifest repository or from
+a local manifest.
 
-With -m, clones the repository at that URL and uses it as the
-manifest repository. If --mr is not given, the remote's default
-branch will be used, if it exists.
-
-With neither, -m {MANIFEST_URL_DEFAULT} is assumed.
-
-Warning: 'west init' renames and/or deletes temporary files inside the
+Arguments
+---------
+--mf
+    The relative path to the manifest file within the manifest repository
+    (remote or local). Config option manifest.file will be set to this value.
+    Defaults to `{_WEST_YML}` if not provided.
+
+--topdir
+    Specifies the directory where west should create the workspace.
+    The `.west` folder will be created inside this directory.
+
+
+1. Remote Manifest Repository (default)
+---------------------------------------
+West clones a remote repository (provided via `-m / --manifest-url`) into a new
+workspace. The repositoy has to contain a west manifest.
+Inside the new workspace, the local config option `manifest.path` is set to
+point at the cloned directory (relative path to the workspace).
+
+If no `-m / --manifest-url` is provided, west uses Zephyr URL by default:
+  {MANIFEST_URL_DEFAULT}.
+
+The west workspace (topdir) is determined as follows:
+- `-t / --topdir` if provided
+- `[directory]` if provided
+- the current working directory (if neither `--topdir` nor `[directory]` are
+  provided)
+
+Note:
+- If both `-t / --topdir` and `[directory]` are provided, `-t / --topdir`
+  specifies the workspace directory, while `[directory]` specifies a subfolder
+  where the manifest repository is cloned inside the workspace.
+- With `--mr`, you can specify the revision (branch, tag, or sha) of the
+  remote repository that will be cloned. It defaults to the repository's
+  default branch (if available).
+
+2. Local Manifest
+-----------------
+If `-l / --local` is given, west initializes a new workspace from an already
+existing local manifest. The directory containing the manifest can be provided
+in positional argument `[manifest_directory]` (it defaults to current working
+directory if no value is provided).
+Inside the new workspace, the local config option `manifest.path` is set to
+point at the directory containing the manifest (relative path to the workspace)
```

Comment:
This is duplicated across 1. and 2. I would move it at the top, just before the "Arguments" section.